app/src/full/java/io/homeassistant/companion/android/sensors/ActivitySensorManager.kt

```diff
@@ -119,16 +190,52 @@ class ActivitySensorManager : BroadcastReceiver(), SensorManager {
     }
 
     override fun requestSensorUpdate(context: Context) {
-        if (!isEnabled(context, activity.id))
-            return
-
-        val actReg = ActivityRecognition.getClient(context)
-        val pendingIntent = getPendingIntent(context)
-        Log.d(TAG, "Unregistering for activity updates.")
-        actReg.removeActivityUpdates(pendingIntent)
-
-        Log.d(TAG, "Registering for activity updates.")
-        actReg.requestActivityUpdates(120000, pendingIntent)
+        if (isEnabled(context, activity.id)) {
+            val actReg = ActivityRecognition.getClient(context)
+            val pendingIntent = getPendingIntent(context)
+            Log.d(TAG, "Unregistering for activity updates.")
+            actReg.removeActivityUpdates(pendingIntent)
+
+            Log.d(TAG, "Registering for activity updates.")
+            actReg.requestActivityUpdates(120000, pendingIntent)
+        }
+        if (isEnabled(context, sleepConfidence.id) || isEnabled(context, sleepSegment.id)) {
+            val pendingIntent = getSleepPendingIntent(context)
+            Log.d(TAG, "Unregistering for sleep updates")
+            ActivityRecognition.getClient(context).removeSleepSegmentUpdates(pendingIntent)
+            Log.d(TAG, "Registering for sleep updates")
+            val task = when {
```

Comment:
Do we really want to unregister then reregister every time we get a requested update?